### ISSUE TITLE
[Feature]: Online video support for VLMs

### ISSUE THREAD

**Issue by litianjian** [vllm-project/vllm#9842]
### Online video support for VLMs
vLLM already supports a large number of MultiModal Machine Learning visual models, some of which support image and video input，such as [Qwen2-VL](https://huggingface.co/Qwen/Qwen2-VL-2B-Instruct), [LLaVA-Onevision](https://huggingface.co/llava-hf/llava-onevision-qwen2-7b-ov-hf), etc. Referring to the implementation of image, this proposal adds support for video.

Refer to the visual interfaces of OpenAI [(vision](https://platform.openai.com/docs/guides/vision)and [video](https://cookbook.openai.com/examples/gpt_with_vision_for_video_understanding)) and Google Gemini, the visual interface should ideally support inputs from Video URLs and base64.

#### Image

```python
chat_response = client.chat.completions.create(
    model="llava",
    messages=[{
        "role": "user",
        "content": [
            {"type": "text", "text": "What’s in this image?"},
            {"type": "image_url", "image_url": {"url": f"data:image/png;base64,{image}"}},
        ],
    }],
)
```

#### Video
```python
chat_response = client.chat.completions.create(
    model="llava",
    messages=[{
        "role": "user",
        "content": [
            {"type": "text", "text": "What’s in this video?"},
            {"type": "video_url", "video_url": {"url": video_url}},
        ],
    }],
)
```

```python
chat_response = client.chat.completions.create(
    model="llava",
    messages=[{
        "role": "user",
        "content": [
            {"type": "text", "text": "What’s in this video?"},
            {"type": "video_url", "video_url": {"url": f"data:video/png;base64,{video_base64}"}},
        ],
    }],
)
```
The interface design mentioned above is a prototype.  I have implemented this interface in my pipeline.

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by DarkLight1337** [comment]
This sounds reasonable. By the way, OpenAI has a rather different way of passing audio files (see their [Audio API](https://platform.openai.com/docs/guides/audio/quickstart?audio-generation-quickstart-example=audio-in)).

```py
completion = client.chat.completions.create(
    model="gpt-4o-audio-preview",
    modalities=["text", "audio"],
    audio={"voice": "alloy", "format": "wav"},
    messages=[
        {
            "role": "user",
            "content": [
                { 
                    "type": "text",
                    "text": "What is in this recording?"
                },
                {
                    "type": "input_audio",
                    "input_audio": {
                        "data": encoded_string,
                        "format": "wav"
                    }
                }
            ]
        },
    ]
)

print(completion.choices[0].message)
```

**Comment by litianjian** [comment]
> This sounds reasonable. By the way, OpenAI has a different way of passing audio files (see their [Audio API](https://platform.openai.com/docs/guides/audio/quickstart?audio-generation-quickstart-example=audio-in)) where the file type inside the base64 URL is stored in `format`.
> 
> ```python
> completion = client.chat.completions.create(
>     model="gpt-4o-audio-preview",
>     modalities=["text", "audio"],
>     audio={"voice": "alloy", "format": "wav"},
>     messages=[
>         {
>             "role": "user",
>             "content": [
>                 { 
>                     "type": "text",
>                     "text": "What is in this recording?"
>                 },
>                 {
>                     "type": "input_audio",
>                     "input_audio": {
>                         "data": encoded_string,
>                         "format": "wav"
>                     }
>                 }
>             ]
>         },
>     ]
> )
> 
> print(completion.choices[0].message)
> ```

This design is indeed clearer.  BTW, does the Vision API need the file type when using base64 URL?

**Comment by DarkLight1337** [comment]
> This design is indeed clearer. BTW, does the Vision API need the file type when using base64 URL?

Yes, Vision API expects the full base64 URL, whereas Audio API expects file type + main base64 content separately.

**Comment by DarkLight1337** [comment]
Let's follow Vision API for now. In the future, we can support both styles for all modalities.

**Comment by alex-jw-brooks** [comment]
I had actually started some prototyping for online video support as well. I agree with the design proposed here - have you already implemented parsing of the video URL? I think it's a bit trickier than images since as far as I'm aware, OpenCV doesn't have an interface for encoding/decoding from memory, so I'm curious about the details of it!

@DarkLight1337 @litianjian In addition to supporting `video_url`, I think it's also a good idea to discuss how to handle adjacent images, since their example treats videos as a sequence of frames. One potential way to handle this is an initialization time flag that controls whether or not adjacent fields that resolve to `image` placeholders should be collapsed into a single `video` placeholder to resolve potential ambiguity for models that support both multi-image and video, and by default, use the same utility to collapse adjacent images for video-only models. I do think that if it's something that makes sense though, it should be a separate PR from supporting `video_url` though
